src/Microsoft.Extensions.Hosting.Abstractions/IApplicationEnvironment.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Provides information about the hosting environment an application is running in.
+    /// </summary>
+    public interface IApplicationEnvironment
+    {
+        /// <summary>
+        /// Gets or sets the name of the environment. This property is automatically set by the host to the value
+        /// of the "DOTNETCORE_ENVIRONMENT" environment variable.
```

Comment:
Note that `Hosting:Environment` is currently marked as [legacy](https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs#L54) in WebHostBuilder. I guess the main reason is the colon but maybe there already was some other discussion regarding the "Hosting" prefix?